inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSink.java

```diff
@@ -311,14 +316,16 @@ private class ConfigChangeProcessor implements Runnable {
         @Override
         public void run() {
             long lastCheckTime;
+            logger.info("{} config-change processor start!", getName());
             while (!isShutdown) {
+                reentrantLock.lock();
```

Comment:
When updating metadata, it is not necessary using lock.

Reply:
This lock with the condition is used to coordinate the message notification between the configuration change notification thread and the configuration update thread in conjunction